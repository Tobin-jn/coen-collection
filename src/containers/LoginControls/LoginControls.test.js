/* eslint-disable */

import React from "react";
import { shallow } from "enzyme";
import {LoginControls, mapStateToProps, mapDispatchToProps} from "./index";

describe("LoginControls", () => {
  let wrapper;
  // let defaultState;

  beforeEach(() => {
    wrapper = shallow(<LoginControls 
      loggedIn={true}
      handleLogin={jest.fn()}
      saveName={jest.fn()}
      location={{pathname: ''}} 
      />)

    // defaultState = {
    //   email: "",
    //   password: "",
    //   username: "",
    //   pathname: this.props.location.pathname,
    //   error: false,
    //   errorMessage: ""
    // };
  })

  it('should exist', () => {
    expect(wrapper).toBeDefined();
  })

  it("should render like snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  // it('should have default state', () => {
  //   expect(JSON.stringify(wrapper.state())).toEqual(JSON.stringify(defaultState));
  // });

  
});