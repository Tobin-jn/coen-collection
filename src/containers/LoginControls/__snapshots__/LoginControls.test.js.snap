// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginControls should render like snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoginControls
    handleLogin={[MockFunction]}
    location={
      Object {
        "pathname": "",
      }
    }
    loggedIn={true}
    saveName={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        false,
        <div
          className="loggedIn"
        >
          <Redirect
            push={false}
            to="/"
          />
        </div>,
        <section>
          <p
            className="error-msg"
          >
            
          </p>
        </section>,
      ],
      "className": "LoginControls",
    },
    "ref": null,
    "rendered": Array [
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Redirect
            push={false}
            to="/"
          />,
          "className": "loggedIn",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "push": false,
            "to": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <p
            className="error-msg"
          >
            
          </p>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "className": "error-msg",
          },
          "ref": null,
          "rendered": "",
          "type": "p",
        },
        "type": "section",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          false,
          <div
            className="loggedIn"
          >
            <Redirect
              push={false}
              to="/"
            />
          </div>,
          <section>
            <p
              className="error-msg"
            >
              
            </p>
          </section>,
        ],
        "className": "LoginControls",
      },
      "ref": null,
      "rendered": Array [
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Redirect
              push={false}
              to="/"
            />,
            "className": "loggedIn",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "push": false,
              "to": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <p
              className="error-msg"
            >
              
            </p>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
              "className": "error-msg",
            },
            "ref": null,
            "rendered": "",
            "type": "p",
          },
          "type": "section",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
